src/core/Akka.Cluster/Routing/ClusterRoutingConfig.cs

```diff
@@ -1054,29 +978,28 @@ public Address SelectDeploymentTarget()
         /// <param name="message">TBD</param>
         protected override void OnReceive(object message)
         {
-            // Moved from RouterPoolActor
-            var poolSize = message as AdjustPoolSize;
-            if (poolSize != null)
+            switch(message)
             {
-                if (poolSize.Change > 0)
-                {
+                // Moved from RouterPoolActor
+                case AdjustPoolSize poolSize when poolSize.Change > 0:
```

Comment:
Did we not support this properly before? @Zetanova

Reply:
What property? 
There should be no behavior changes in this commit. 
Only some cleanup.

Reply:
Oh I see what you mean - duh. I read the diff wrong.